src/ui/common/src/components/workflows/workflowHeader.tsx

```diff
@@ -89,17 +91,45 @@ const WorkflowHeader: React.FC<Props> = ({ user, workflowDag }) => {
     [key: string]: string;
   }>({});
 
+  // Returns the map of parameters, from name to serialized spec (which includes the
+  // base64-encoded value and serialization_type).
+  const serializeParameters = () => {
+    let serializedParams = {}
+    Object.entries(paramNameToValMap).forEach(([key, strVal]) => {
+      // Serialize the user's input string appropriately into base64. The input can either be a
+      // 1) number 2) string 3) json.
+      try {
+        let val = JSON.parse(strVal)
+
+        // All jsonable values are serialized as json.
+        serializedParams[key] = JSON.stringify({
+          val: btoa(JSON.stringify(val)),
+          serialization_type: SerializationType.Json,
+        })
```

Comment:
What matters is just that the serialization type matches the serialization of the value. In the SDK, we serialize JSON strings with SERIALIZATION_TYPE.STRING, but here we can json serialize JSON strings! As long as the serialization type matches, the backend can always infer the correct type.

The reason is because the SDK and executor use `infer_artifact_type()`, and then use this inferred type to index into a map. However, the frontend is different because the input is always a string, so really the value is either JSONable or not. The user can still enter a raw json string eg `"{\"hello\": \"world\"}"` and things will work just fine.